### PR TITLE
Make DisplayOrder effective and items sorted by ASC

### PR DESCRIPTION
**Pull request by gdlcf88**
https://github.com/EasyAbp/EasyAbpGuide/issues/19

### CODE REVIEW

File: modules/EasyAbp.EShop.Products/src/EasyAbp.EShop.Products.Application/EasyAbp/EShop/Products/ProductCategories/ProductCategoryAppService.cs [hunk 1]
```diff
@@ -34,6 +34,8 @@ protected override async Task<IQueryable<ProductCategory>> CreateFilteredQueryAs
                 queryable = queryable.Where(x => x.ProductId == input.ProductId);
             }
 
+            queryable = queryable.OrderBy(x => x.DisplayOrder);
```
Comment (JadynWong) [inline modules/EasyAbp.EShop.Products/src/EasyAbp.EShop.Products.Application/EasyAbp/EShop/Products/ProductCategories/ProductCategoryAppService.cs:37]: It is better to sort in `ApplySorting`. Is it expected to always sort by CreationTime when `Input.Sorting` is empty?

https://github.com/abpframework/abp/blob/edda93473dd458af31be82ec55cd371ec2a952ce/framework/src/Volo.Abp.Ddd.Application/Volo/Abp/Application/Services/AbstractKeyReadOnlyAppService.cs?#L93-L131

File: modules/EasyAbp.EShop.Products/src/EasyAbp.EShop.Products.Application/EasyAbp/EShop/Products/Products/ProductAppService.cs [hunk 2]
```diff
@@ -61,7 +61,8 @@ protected override async Task<IQueryable<Product>> CreateFilteredQueryAsync(GetP
             return query
                 .Where(x => x.StoreId == input.StoreId)
                 .WhereIf(!input.ShowHidden, x => !x.IsHidden)
-                .WhereIf(!input.ShowUnpublished, x => x.IsPublished);
+                .WhereIf(!input.ShowUnpublished, x => x.IsPublished)
+                .OrderBy(x => x.DisplayOrder);
```
Comment (JadynWong) [inline modules/EasyAbp.EShop.Products/src/EasyAbp.EShop.Products.Application/EasyAbp/EShop/Products/Products/ProductAppService.cs:65]: Same as above.

File: modules/EasyAbp.EShop.Products/src/EasyAbp.EShop.Products.Application/EasyAbp/EShop/Products/Products/ProductViewAppService.cs [hunk 3]
```diff
@@ -51,7 +51,8 @@ protected override async Task<IQueryable<ProductView>> CreateFilteredQueryAsync(
             return query
                 .Where(x => x.StoreId == input.StoreId)
                 .WhereIf(!input.ShowHidden, x => !x.IsHidden)
-                .WhereIf(!input.ShowUnpublished, x => x.IsPublished);
+                .WhereIf(!input.ShowUnpublished, x => x.IsPublished)
+                .OrderBy(x => x.DisplayOrder);
```
Comment (JadynWong) [inline modules/EasyAbp.EShop.Products/src/EasyAbp.EShop.Products.Application/EasyAbp/EShop/Products/Products/ProductViewAppService.cs:55]: Same as above.

File: modules/EasyAbp.EShop.Products/src/EasyAbp.EShop.Products.Application/EasyAbp/EShop/Products/ProductCategories/ProductCategoryAppService.cs (hunk 1 above)
Comment (gdlcf88) [reply modules/EasyAbp.EShop.Products/src/EasyAbp.EShop.Products.Application/EasyAbp/EShop/Products/ProductCategories/ProductCategoryAppService.cs]: I Think the sequential is enough. CreationTime sorting needs additional indexes.

**Review by JadynWong — APPROVED** [review]

**Pull request merged**